File: grind75/Dec20/57.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) 
-> List[List[int]]:
        new_intervals = []
        inserted = False
        for i, interval in enumerate(intervals):
            # before the overlap
            if interval[1] < newInterval[0]:
                new_intervals.append(interval)
            elif interval[0] > newInterval[1]:
                    new_intervals.append(newInterval)
                    return new_intervals + intervals[i:]
            else:
                # in overlap
                newInterval = [min(interval[0], newInterval[0]), 
max(interval[1], newInterval[1])]
        
        new_intervals.append(newInterval)
        return new_intervals

File: grind75/Dec20/57.go
func insert(intervals [][]int, newInterval []int) [][]int {
    ledge, redge := newInterval[0], newInterval[1]
    l := len(intervals)
    if l == 0 {
        return [][]int{[]int{ledge, redge}}
    }
    ret := [][]int{}
    i := 0
    for i < l && intervals[i][1] < newInterval[0] {
        ret = append(ret, intervals[i])
        i++
    }

    for i < l && intervals[i][0] <= newInterval[1] {
        newInterval[0] = min(intervals[i][0], newInterval[0])
        newInterval[1] = max(intervals[i][1], newInterval[1])
        i++
    }
    ret = append(ret, newInterval)

    for i < l {
        ret = append(ret, intervals[i])
        i++
    }
    return ret
}
